hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/protocolPB/OzoneManagerRequestHandler.java

```diff
@@ -1219,11 +1219,26 @@ private GetS3VolumeContextResponse getS3VolumeContext()
 
   private SnapshotDiffResponse snapshotDiff(
       SnapshotDiffRequest snapshotDiffRequest) throws IOException {
-    return SnapshotDiffResponse.newBuilder().setSnapshotDiffReport(
-        impl.snapshotDiff(snapshotDiffRequest.getVolumeName(),
+    org.apache.hadoop.ozone.snapshot.SnapshotDiffResponse response =
+        impl.snapshotDiff(
+            snapshotDiffRequest.getVolumeName(),
             snapshotDiffRequest.getBucketName(),
             snapshotDiffRequest.getFromSnapshot(),
-            snapshotDiffRequest.getToSnapshot()).toProtobuf()).build();
+            snapshotDiffRequest.getToSnapshot(),
+            snapshotDiffRequest.getToken(),
+            snapshotDiffRequest.getPageSize());
+
+    OzoneManagerProtocolProtos.SnapshotDiffReportProto snapshotDiffReportProto;
+    if (response.getSnapshotDiffReport() == null) {
+      snapshotDiffReportProto = null;
+    } else {
+      snapshotDiffReportProto = response.getSnapshotDiffReport().toProtobuf();
+    }
+
+    return SnapshotDiffResponse.newBuilder()
+        .setSnapshotDiffReport(snapshotDiffReportProto)
```

Comment:
Double check if setting `null` would cause protobuf-generated methods to raise exception or not? iirc this could be a potential issue with proto3 (tho I believe we are using protoc 3 to compile proto2 syntax). Do we have a test case for this?